### PR TITLE
Validate workDir and userName parameters

### PR DESCRIPTION
**Pull request by jordeu**
## Description

- Validate the `userName` at start time and report error if not possible to detect.
- Check that `workDir` is valid and exists at start time. Report and error if not.  
- Log agent version, API required version, `userName` and `workDir` at startup.

### CODE REVIEW

File: src/main/java/io/seqera/tower/agent/Agent.java [hunk 1]
```diff
@@ -182,17 +188,47 @@ private void sendPeriodicHeartbeat() {
...
+        }
+
+        // Set default workDir
+        if (workDir == null) {
+            String defaultPath = System.getProperty("user.home") + "/work";
+            try {
+                workDir = Paths.get(defaultPath);
+            } catch (InvalidPathException e) {
+                logger.error("Impossible to define a default work directory. Please provide one using '--work-dir'.");
+                System.exit(-1);
```
Comment (pditommaso) [inline src/main/java/io/seqera/tower/agent/Agent.java:213]: -1 is weird. It should be 1, no ?

File: src/main/java/io/seqera/tower/agent/Agent.java [hunk 2]
```diff
@@ -182,17 +188,47 @@ private void sendPeriodicHeartbeat() {
...
+        // Fetch username
+        validatedUserName = System.getProperty("user.name");
+        if (validatedUserName == null || validatedUserName.isEmpty() || validatedUserName.isBlank()) {
+            logger.error("Impossible to detect current Unix username.");
+            System.exit(-1);
+        }
+
+        // Set default workDir
+        if (workDir == null) {
+            String defaultPath = System.getProperty("user.home") + "/work";
```
Comment (pditommaso) [inline src/main/java/io/seqera/tower/agent/Agent.java:208]: let's log it conditon

**Review by pditommaso — CHANGES_REQUESTED** [review]
A few comments

File: src/main/java/io/seqera/tower/agent/Agent.java (hunk 1 above)
Comment (jordeu) [reply src/main/java/io/seqera/tower/agent/Agent.java:213]: I've always seen -1 (=255) on error but let's use 1 as we are doing on Tower CLI.

File: src/main/java/io/seqera/tower/agent/Agent.java (hunk 2 above)
Comment (jordeu) [reply src/main/java/io/seqera/tower/agent/Agent.java:208]: What do you mean? Now I always log the `userName` and the `workDir`.
http://github.com/seqeralabs/tower-agent/blob/1c9ef357e3d26e67006b9c9921d36eee39e9d3e9/src/main/java/io/seqera/tower/agent/Agent.java#L229-L229

Comment (pditommaso) [reply src/main/java/io/seqera/tower/agent/Agent.java:208]: was just suggesting to log the `workDir == null` condition. it can be useful for debug (?)

Comment (jordeu) [reply src/main/java/io/seqera/tower/agent/Agent.java:214]: Ok, added as a debug log.

**Review by pditommaso — APPROVED** [review]

**Pull request merged**